src/AbstractPathMappingRepository.php

```diff
@@ -278,4 +299,54 @@ protected function createVirtualResource($path = null)
 
         return $resource;
     }
+
+    /**
+     * Transform a relative path into an absolute path.
+     *
+     * @param string $absolutePath
+     *
+     * @return string
+     */
+    protected function makePathRelative($absolutePath)
+    {
+        if (0 === strpos($absolutePath, 'l:')) {
+            // Link
+            return $absolutePath;
+        }
+
+        return Path::makeRelative($absolutePath, $this->baseDirectory);
+    }
+
+    /**
+     * Transform a relative path into an absolute path.
+     *
+     * @param string $relativePath
+     *
+     * @return string
+     */
+    protected function makePathAbsolute($relativePath)
+    {
+        if (0 === strpos($relativePath, 'l:')) {
+            // Link
+            return $relativePath;
+        }
+
+        return Path::makeAbsolute($relativePath, $this->baseDirectory);
+    }
+
+    /**
+     * Transform a collection of relative paths into a collection of absolute paths.
+     *
+     * @param array $relativePaths
+     *
+     * @return array
+     */
+    protected function makePathsCollectionAbsolute($relativePaths)
+    {
+        foreach ($relativePaths as &$relativePath) {
```

Comment:
I'd rather include the index in the loop and then assign by index rather than by reference. Using references in loops can cause bugs that are hard to spot.